ttnn/cpp/ttnn/operations/eltwise/unary/unary_composite.hpp

```diff
@@ -1,120 +1,45 @@
-        }
+        std::optional<Tensor> optional_output_tensor = std::nullopt);
 
     static Tensor invoke(
         const Tensor& input_tensor,
         float exponent,
         const std::optional<MemoryConfig>& memory_config = std::nullopt,
-        std::optional<Tensor> optional_output_tensor = std::nullopt)
-        {
-        return OpHandler<UnaryCompositeOpType::POW>::handle(DefaultQueueId, input_tensor, exponent, memory_config.value_or(input_tensor.memory_config()), optional_output_tensor);
-        }
-};
-
-template <UnaryCompositeOpType unary_comp_op_type>
-struct ExecuteUnaryCompositeOp {
-    static Tensor invoke(
-        const Tensor& input_tensor, const std::optional<MemoryConfig>& memory_config = std::nullopt) {
-        auto output_memory_config = memory_config.value_or(input_tensor.memory_config());
-        return OpHandler<unary_comp_op_type>::handle(input_tensor, output_memory_config);
-    }
-};
-
-//OpHandler_float : get_function_type_float
-template <UnaryCompositeOpType unary_comp_op_type>
-struct ExecuteUnaryCompositeOpWithFloat {
-
-    //Type : 1 inputs, 1 float
-    static ttnn::Tensor invoke(
-        const Tensor &input_tensor,
-        float param1,
-        const std::optional<MemoryConfig> &memory_config = std::nullopt) {
-        return OpHandler<unary_comp_op_type>::handle(input_tensor, param1, memory_config);
-        }
-};
-
-template <UnaryCompositeOpType unary_comp_op_type>
-struct ExecuteUnaryCompositeOpWithDim
-{
-    static Tensor invoke(
-        const Tensor& input_tensor,
-        int32_t dim,
-        const std::optional<MemoryConfig>& memory_config = std::nullopt)
-        {
-            auto output_memory_config = memory_config.value_or(input_tensor.memory_config());
-            return OpHandler<unary_comp_op_type>::handle(input_tensor, dim, output_memory_config);
-        }
-};
-
-template <UnaryCompositeOpType unary_comp_op_type>
-struct ExecuteUnaryCompositeOpWithFloats {
-    //Type 1: 1 inputs, 2 float
-    static Tensor invoke(
-        const Tensor &input_tensor,
-        float param1,
-        float param2,
-        const std::optional<MemoryConfig> &memory_config = std::nullopt)
-        {
-            auto output_memory_config = memory_config.value_or(input_tensor.memory_config());
-            return OpHandler<unary_comp_op_type>::handle(input_tensor, param1, param2, output_memory_config);
-        }
-};
-
-template <UnaryCompositeOpType unary_comp_op_type>
-struct ExecuteUnaryCompositeOpWithInt {
-
-    static Tensor invoke(
-        const Tensor &input_tensor,
-        int32_t param1,
-        const std::optional<MemoryConfig> &memory_config = std::nullopt) {
-        auto output_memory_config = memory_config.value_or(input_tensor.memory_config());
-        return OpHandler<unary_comp_op_type>::handle(input_tensor, param1, output_memory_config);
-        }
+        std::optional<Tensor> optional_output_tensor = std::nullopt);
 };
 
-struct ExecuteRdiv {
+struct RdivOperation {
```

Comment:
why do you need to have RdivOperation separately?
Is it because of the round_mode?